resource-types/typescript-example-website-monitor/src/handlers.ts

```diff
@@ -0,0 +1,316 @@
+
+    /**
+     * CloudFormation invokes this handler when the resource is deleted, either when
+     * the resource is deleted from the stack as part of a stack update operation,
+     * or the stack itself is deleted.
+     *
+     * @param session Current AWS session passed through from caller
+     * @param request The request object for the provisioning request passed to the implementor
+     * @param callbackContext Custom context object to allow the passing through of additional
+     * state or metadata between subsequent retries
+     * @param logger Logger to proxy requests to default publishers
+     */
+    @handlerEvent(Action.Delete)
+    public async delete(
+        session: Optional<SessionProxy>,
+        request: ResourceHandlerRequest<ResourceModel>,
+        callbackContext: CallbackContext,
+        logger: LoggerProxy
+    ): Promise<ProgressEvent> {
+        logger.log('request', request);
+        const model = new ResourceModel(request.desiredResourceState);
+
+        // The Id property, being the primary identifier, cannot be left empty.
+        if (!model.id) {
+            throw new exceptions.NotFound(this.typeName, request.logicalResourceIdentifier);
+        }
+
+        // Remove the synthetics monitor by calling the delete endpoint with its ID.
+        // https://docs.newrelic.com/docs/apis/synthetics-rest-api/monitor-examples/manage-synthetics-monitors-rest-api#delete-monitor
+        model.endpointRegion = model.endpointRegion || 'US';
+        const apiEndpoint = ApiEndpoints[model.endpointRegion as EndpointRegions];
+        const response: Response = await fetch(`${apiEndpoint}/v3/monitors/${model.id}`, {
+            method: 'DELETE',
+            headers: { ...Resource.DEFAULT_HEADERS, 'Api-Key': model.apiKey }
+        });
+        await this.checkResponse(response, logger, model.id);
+
+        const progress = ProgressEvent.success<ProgressEvent<ResourceModel>>();
+        logger.log('progress', progress);
+        return progress;
+    }
+
+    /**
+     * CloudFormation invokes this handler as part of a stack update operation when
+     * detailed information about the resource's current state is required.
+     *
+     * @param session Current AWS session passed through from caller
+     * @param request The request object for the provisioning request passed to the implementor
+     * @param callbackContext Custom context object to allow the passing through of additional
+     * state or metadata between subsequent retries
+     * @param logger Logger to proxy requests to default publishers
+     */
+    @handlerEvent(Action.Read)
+    public async read(
+        session: Optional<SessionProxy>,
+        request: ResourceHandlerRequest<ResourceModel>,
+        callbackContext: CallbackContext,
+        logger: LoggerProxy
+    ): Promise<ProgressEvent> {
+        logger.log('request', request);
+        const model = new ResourceModel(request.desiredResourceState);
+
+        if (!model.id) {
```

Comment:
Just a general comment here. I don't believe this should have to be done in the code. The service shouldn't be passing something to read that doesn't have a `Id` since its a `primaryIdentifier`

Reply:
Needed for contract tests